test/core/security/security_connector_test.cc

```diff
@@ -29,6 +29,7 @@
 #include "src/core/lib/gpr/tmpfile.h"
 #include "src/core/lib/security/context/security_context.h"
 #include "src/core/lib/security/security_connector/security_connector.h"
+#include "src/core/lib/security/security_connector/security_connector.cc"
```

Comment:
There are a few options here regarding `ssl_host_matches_name`:

1. import the source file and test the static function
2. add a proxy to the header in the test section
3. remove static and promote the function itself to the public test api in the header
4. move the logic to `tsi` function

4 seems best but means it's behind the tsi interface now so some future other implementation of the interface would have to also include this logic.. could be a feature. This is also the most expensive option because it incurs an allocation. And, the logic in `tsi_ssl_peer_matches_name` is more fragile.